server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -298,22 +299,32 @@ public void close() {
     }
 
     public NodeEnvironment(Settings settings, Environment environment) throws IOException {
-        this(settings, environment, IndexStoreListener.EMPTY);
+        this(settings, environment, org.opensearch.index.store.IndexStoreListener.EMPTY);
+    }
+
+    /**
+     * Use {@link NodeEnvironment#NodeEnvironment(Settings, Environment, org.opensearch.index.store.IndexStoreListener)} instead.
+     * This constructor is kept around on 2.x to avoid breaking changes.
+     */
+    @Deprecated(forRemoval = true, since = "2.19.0")
+    public NodeEnvironment(Settings settings, Environment environment, IndexStoreListener indexStoreListener) throws IOException {
```

Comment:
@reta was this change required? I thought we exempted the constructors from the check?

Reply:
> @reta was this change required? I thought we exempted the constructors from the check?

@andrross not by default, but if annotated with `@InternalApi` (not the case)